walkthroughs/1-on-walkthroughs/walkthrough.adoc

```diff
@@ -0,0 +1,407 @@
+
+=== Adding an image to the template
+
+. Create the *images* subdirectory inside your Walkthrough directory
+
+. Copy a .png or .jpg image into that directory
+
+. Add the image to the preamble:
++
+----
+image::images/<name>[diagram, role="integr8ly-img-responsive"]
+----
+
+. Save, commit and push your changes. Then trigger a redeployment of the Webapp like in the previous task.
+
+[type=verification]
+Check that the image is displayed in the Walkthrough overview.
+
+[type=verificationFail]
+Check that the Webapp has finished redeployment and no errors occurred.
+
+[time=5]
+== Adding task resources
+
+Task resources, like walkthrough resources, are used to display helpful information to the user on the right hand side of the Webapp. The difference is that, while Walkthrough resources are displayed on every task, task resources only appear in the task they were defined in.
+
+They are added to task sections but don't have the `serviceName` attribute:
+
+[source,asciidoc]
+----
+[type=taskResource]
+===
+* link:{openshift-host}[Openshift Console]
+----
+
+=== Adding a task resource to the template
+
+. At the end of your Task section, add the following:
++
+----
+[type=taskResource]
+===
+* link:https://google.com[Task related link]
+----
+
+. Save, commit and push your changes. Then trigger a redeployment of the Webapp like in the previous task.
+
+[type=verification]
+Check that the task resource appears on your task, but not on the overview page.
+
+[type=verificationFail]
+Check that the Webapp has finished redeployment and no errors occurred.
+
+[time=10]
+== Adding verifications
+
+Every Procedure in a Task should have a verification. Verifications are used to make sure that the steps taken by the user were successful. They should be formulated as questions and the user can answer them with `Yes` or `No`.
+
+As with resources, block attributes are used to define verifications
+
+=== Adding a verification to our task
+
+. At the end of the Procedure section, right after the numbered list add:
++
+[source,asciidoc]
+----
+[type=verification]
+Check that the dashboard of service X reports no errors.
+----
+
+. To provide a better feedback to users when something didn't go as planned you should use *verificationFail* blocks. They allow you to display text that will be displayed when the users answers the verification with `No`:
+
+. After the verification block add:
++
+[source,asciidoc]
+----
+[type=verificationFail]
+Try turning it off and on again.
+----
+
+. Save, commit and push your changes. Then trigger a redeployment of the Webapp like in the previous task.
+
+[type=verification]
+Check that the verification appears on the task. Check that you can click the *Next* button when answered *Yes* and that the fail text appears when answering *No*.
+
+[type=verificationFail]
+Check that the verification has been added after the Procedure section. It will be ignored in any other location.
+
+[time=5]
+== Using attributes
```

Comment:
@aidenkeating @finp  I feel we should add more content here but i'm not sure how to assist the user: is there an easy way to display all the available attributes?

Reply:
@pb82 At the moment there's not really a good way to display to the end-user all of the attributes available to them. We could, for now, manually list all of the attributes that we know will exist, but that won't include dynamic attributes like routes that are found.

https://github.com/integr8ly/example-customisations/blob/master/walkthroughs/my-custom-walkthrough/walkthrough.adoc#using-walkthrough-attributes

Reply:
Thanks, i'll add them for now